2-0-data-structures-and-algorithms/src/main/java/com/bobocode/linked_queue/LinkedQueue.java

```diff
@@ -0,0 +1,49 @@
+package com.bobocode.linked_queue;
+
+import com.bobocode.util.ExerciseNotCompletedException;
+
+/**
+ * {@link LinkedQueue} implements FIFO {@link Queue}, using singly linked nodes. Nodes are stores in instances of nested
+ * class Node. In order to perform operations {@link LinkedQueue#add(Object)} and {@link LinkedQueue#poll()}
+ * in a constant time, it keeps to references to the head and tail of the queue.
+ *
+ * @param <T> a generic parameter
+ */
+public class LinkedQueue<T> implements Queue<T> {
+
+    /**
+     * Adds an element to the end of the queue.
+     *
+     * @param element the element to add
+     */
+    public void add(T element) {
+        throw new ExerciseNotCompletedException(); // todo: implement this method
```

Comment:
@erneman1 completed brach **should contain a final solution** which means all methods should be implemented and all tests should pass.